src/shared/logger.ts

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as path from 'path'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+import * as winston from 'winston'
+
+const localize = nls.loadMessageBundle()
+let logSettings: LogSettings
+
+class LogSettings {
```

Comment:
Consider making `LogSettings`  an interface with a corresponding `getLogger()` that returns something that perhaps implements logging interface such as `ILogService` or an extension of it. 

It's worth trying out. All of the `private readonly` declarations disappear. No more` this.<propertyName> = <propertyName>`. It might like an unexpected approach but it's often a cleaner more flexible solution than a class. Using a factory `getLogger()` could also support the singleton pattern if we decide it's desirable.

Reply:
I'm not 100% sure if this will work with Winston since I don't want to risk overwriting existing output channels or log files. I can look into it though. See my notes re: ILogService below, however; that is not currently an option.